### PR TITLE
Extract a RedirectionCreation service

### PR DESCRIPTION
**Pull request by gabebw**
RedirectionCreation handles creating (or not creating!) a new Redirection based on a slug and referrer. It also links the new redirection into the ring.

Since we handle edge cases (like a blank referrer) at the RedirectionCreation unit test level, the controller specs can now be correspondingly simpler.

### CODE REVIEW

File: app/services/redirection_creation.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      redirection
+    end
+  end
+
+  private
+
+  attr_reader :referrer, :slug
+
+  def referrer_hostname
+    referrer_uri = URI.parse(referrer)
```
Comment (edwardloveall) [inline app/services/redirection_creation.rb]: I only knew about `URI(referrer)`. `.parse` is probably slightly more clear anyway.

File: app/services/redirection_creation.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class RedirectionCreation
+  def initialize(referrer, slug)
+    @referrer = referrer
+    @slug = slug
+  end
+
+  def perform
```
Comment (edwardloveall) [inline app/services/redirection_creation.rb]: I'd also recommend a class method to go along side this:

```ruby
def self.perform(referrer, slug)
  new(referrer, slug).perform
end
```

because it makes future testing (stubbing in particular) much easier. I.e. you don't have to stub `new` to return a double then stub out `perform` on the double. Also, if for whatever reason we use background jobs, I hear having those `perform` methods can make that easier too.

**Comment by edwardloveall** [comment]
Looks pretty good to me :red_circle: :black_circle: :white_circle: :large_blue_circle:

File: app/services/redirection_creation.rb (hunk 2 above)
Comment (gabebw) [inline app/services/redirection_creation.rb]: Done!

**Comment by edwardloveall** [comment]
Yeah it's lookin good in this neighborhood

File: app/services/redirection_creation.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class RedirectionCreation
+  def self.perform(referrer, slug)
+    new(referrer, slug).perform
```
Comment (adarsh) [inline app/services/redirection_creation.rb]: Beauty!
